specification/security/resource-manager/Microsoft.Security/preview/2024-03-01-preview/securityConnectors.json

```diff
@@ -0,0 +1,1618 @@
+            "cloudRoleArn": {
+              "type": "string",
+              "description": "The cloud role ARN in AWS for this feature used for reading data"
+            }
+          }
+        },
+        "cloudWatchToKinesis": {
+          "type": "object",
+          "description": "The cloudwatch to kinesis connection configuration",
+          "properties": {
+            "cloudRoleArn": {
+              "type": "string",
+              "description": "The cloud role ARN in AWS used by CloudWatch to transfer data into Kinesis"
+            }
+          }
+        },
+        "kinesisToS3": {
+          "type": "object",
+          "description": "The kinesis to s3 connection configuration",
+          "properties": {
+            "cloudRoleArn": {
+              "type": "string",
+              "description": "The cloud role ARN in AWS used by Kinesis to transfer data into S3"
+            }
+          }
+        },
+        "enableAuditLogsAutoProvisioning": {
+          "type": "boolean",
+          "description": "Is audit logs data collection enabled"
+        },
+        "enableDefenderAgentAutoProvisioning": {
+          "type": "boolean",
+          "description": "Is Microsoft Defender for Cloud Kubernetes agent auto provisioning enabled"
+        },
+        "enablePolicyAgentAutoProvisioning": {
+          "type": "boolean",
+          "description": "Is Policy Kubernetes agent auto provisioning enabled"
+        },
+        "kubeAuditRetentionTime": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The retention time in days of kube audit logs set on the CloudWatch log group"
+        },
+        "dataCollectionExternalId": {
+          "type": "string",
+          "description": "The externalId used by the data reader to prevent the confused deputy attack"
+        },
+        "mdcContainersImageAssessment": {
+          "type": "object",
+          "description": "The Microsoft Defender container image assessment configuration",
+          "properties": {
+            "enabled": {
+              "type": "boolean",
+              "description": "Is Microsoft Defender container image assessment enabled"
+            },
+            "cloudRoleArn": {
+              "type": "string",
+              "description": "The cloud role ARN in AWS for this feature"
+            }
+          }
+        },
+        "mdcContainersAgentlessDiscoveryK8s": {
+          "type": "object",
+          "description": "The Microsoft Defender container agentless discovery K8s configuration",
+          "properties": {
+            "enabled": {
+              "type": "boolean",
+              "description": "Is Microsoft Defender container agentless discovery K8s enabled"
+            },
+            "cloudRoleArn": {
+              "type": "string",
+              "description": "The cloud role ARN in AWS for this feature"
+            }
+          }
+        },
+        "vmScanners": {
```

Comment:
This is strange, `vmScanners` has allOf[vmScannersAws], and `vmScannersAws` has allof [vmScanners]? Please check and fix the definitions

Reply:
hey, the vmscanner is a configuration which can be applied to 3 different offerings across 2 environments. the configuration is the same for each cloud - aws and gcp. to avoid code duplication, i created a "vmscanners" object (consider this vmscannerBase) and 2 inheriting objects - vmscannerAWs and vmScannerGCP. 

The vmScanner **property** in the actual resource body can be of type vmscannerAws or vmScannerGCp. 
perhaps renaming vmScanner to vmscannerBase would help explain that the object is a base class and the property is a reference to one of the inheriting classes.. does that make sense?